### ISSUE TITLE
[Feature Request] DFT/STFT WebGPU op support

### ISSUE THREAD

**Issue by xenova** [microsoft/onnxruntime#20997]
### Describe the feature request

To add WebGPU support for DFT and STFT ops:
- https://onnx.ai/onnx/operators/onnx__STFT.html
- https://onnx.ai/onnx/operators/onnx__DFT.html

Currently, neither are supported yet (as stated [here](https://github.com/microsoft/onnxruntime/blob/main/js/web/docs/webgpu-operators.md)).

### Describe scenario use case

This will be used to optimize preprocessing performance for audio models like Whisper, specifically to be used in Transformers.js.

**Comment by xenova** [comment]
cc @guschmue for this. Another thing I've noticed is that the WASM version is much slower than a pure JS implementation (which I currently use). So, might be something to look into too.

**Comment by guschmue** [comment]
makes sense. Kind surprising that wasm is slower then javascript, maybe it's an algo choice. We'll look at it.
